File: 2/project-euler-problem2.js
function fiboEvenSum(limit) {
    var fibCache = {
        "1": 1,
        "2": 2
    }

    var fib = (x) => {
        if (Boolean(fibCache[x]) === false) {
            fibCache[x] = fib(x - 1) + fib(x - 2)
        }
        return fibCache[x]
    }

    var sum = 0;
    for (let n = 1; fib(3 * n - 1) <= limit; n++) {
        sum += fib(3 * n - 1);
    }

    return sum
}

fiboEvenSum(10);

File: 2/project-euler-problem2.py
def fiboEvenSum(limit):
    fibCache = {
        "1": 1,
        "2": 2
    }

    def fib(x):
        if (str(x) in fibCache) == False:
            fibCache[str(x)] = fib(x - 1) + fib(x - 2)
        return fibCache[str(x)]

    sum = 0
    n = 1
    while (fib(3 * n - 1) <= limit):
        sum = sum + fib(3 * n - 1)
        n = n + 1
    return sum

print(fiboEvenSum(4000000))
# output: 4613732
